### ISSUE TITLE
Date timezone doesnt adhere to browsers/servers timezone

### ISSUE THREAD

**Issue by gregg-cbs** [strapi/strapi#18858]
## Bug report
When i save a date in strapi as some date and time, the time being sent to the server is different to the time i see in the browser - its not adhering to my timezone. This is happening on my local. The time should be the same timezone that I am in. This issue created this problem: https://github.com/strapi/strapi/issues/8739

### Required System information
- Node.js version: 18
- NPM version: 6
- Strapi version: 4.15.0
- Database: Postgres
- Operating system: Windows
- Is your project Javascript or Typescript: TS

### Describe the bug
I run strapi locally on my OS. I go into the strapi cms and update one of my collection items fields dates.
I set the date to: "02 Dec 2023 and the time to 9am".
The network request sends the time as UTC "2023-12-02T07:00:00.000Z"

Now if I want to always show the time as 9am not matter the timezone, how do i do this?

### Steps to reproduce the behavior
1. Go to any date time field
2. Set the date and time
3. Look at network request
4. See time is not the timezone you are on

### Expected behavior
In this case i expect the time to be saved as my local time "2023-12-02T09:00:00...Z" and not some other timezone. My server and browser is running on my pc which is South African timezone but im seeing UTC timezone?

### Screenshots
Time saved in UI:
![image](https://github.com/strapi/strapi/assets/57114335/706b3173-438c-4c4c-87ab-d3e02d65cbcd)

Time sent in request:
![image](https://github.com/strapi/strapi/assets/57114335/f291d4c1-3c58-4f75-95a5-6b06a9e99b59)

Web browser timezone (https://webbrowsertools.com/timezone/)
![image](https://github.com/strapi/strapi/assets/57114335/0aadb4f5-21c0-4b41-a13d-a3e3863a01e2)

**Comment by gregg-cbs** [comment]
And i get the argument for https://github.com/strapi/strapi/issues/8739 but solving that has now opened up this issue.

**Comment by smoothdvd** [comment]
You should set your db environment with proper timezone

**Comment by derrickmehaffy** [comment]
> You should set your db environment with proper timezone

Absolutely not, databases should be using UTC always. The frontend should be doing the conversion

**Comment by derrickmehaffy** [comment]
Admin (Local TZ convert to UTC) ---- sent to backend ---> Backend(UTC) ---> Database (UTC)

Database (UTC) ----> Backend (UTC) ---> Frontend (Convert to local TZ)

**Comment by gregg-cbs** [comment]
I get what you saying @derrickmehaffy and that is correct if you want the conversion to happen. I do not want the conversion to happen.

How would i show the time as 9am in every country? I save the time as 9am in my browser, you convert it to UTC and it gets saved to the DB as UTC - 7am, I want to show it as 9am everywhere?

**Comment by derrickmehaffy** [comment]
Yeah sadly for the moment we don't have that but I've pushed to have its priority raised but we need more people who need that use-case to make some noise so I can show that to my product team.

**Comment by nelsonmiew** [comment]
Hello, any updates related to that or any workaround? 
I'm having the same issue too.

**Comment by gregg-cbs** [comment]
@nelsonmiew Im kinda of lucky because I only need the dates to be in 1 timezone. My server is in the UK so strapi saves everything as uk time. My clients are in South Africa so i need to show it as South African time.

Im going to modify the datetime in the `beforeCreate` and `beforeUpdate` so if someone saves the time as 9am and strapi changes it to UTC 7am, im going to change the 7 to 9 and when parsing dates I wont use the timezone so it keeps it as 9am.

Its complicated because my frontend is react nextjs which renders serverside and clientside - if the server is in a different timezone to the frontend then the user is going to see different times depending on whether the page is rendered serverside or clientside. Thats why strapis chosen utc date format is not great.

The date format I am hoping strapi changes to is this:
"2023-12-20T16:00:00+02:00"

**Comment by gregg-cbs** [comment]
@nelsonmiew - I am looking into this today to see if I can find a solution myself. So far no luck.

**Comment by gregg-cbs** [comment]
@nelsonmiew 
So i can set the date fields in the database to have/save the format that I want:
![image](https://github.com/strapi/strapi/assets/57114335/ccfcdb8e-2f84-4828-83d9-4947b0ff1c41)

But strapi always converts it at query time to be this format 🤦‍♂️
![image](https://github.com/strapi/strapi/assets/57114335/3be08695-66f1-4d71-821a-2ce1e8c6e22d)

**Comment by gregg-cbs** [comment]
@nelsonmiew i have a solution.

I have 2 date fields, date_start and date_end. I will leave those as is. I then created 2 string fields date_start_timezone and date_end_timezone that I manually set in the lifecycle hooks `beforeCreate` and `beforeUpdate` to the format that I want and I use these fields instead. I will hide these fields from strapi so the user doesnt see them but use them in the system.

I thought I would be clever and rename my old fields to something else and create the new fields in the name of the old fields to avoid having to make code changes but renaming a field in strapi wipes all its data 🤦‍♂️ so dont do that.

![image](https://github.com/strapi/strapi/assets/57114335/ec839f63-6859-4c97-8d0f-8f102fa055de)

**Comment by Scalpeletto** [comment]
Just chiming in to say I searched google for this issue because I've found the Time datetime content types to be absolutely useless without timezone handling.

**Comment by Drarok** [comment]
Not sure if I'm experiencing the same or a related bug;

I have a DateTime field and set its value to `2009-09-18 01:10` using the content manager. This is sent to the server as `2009-09-18T00:10:00.000Z` when I press save, and returned from the API as `2009-09-18T00:10:00.000Z` too.

I'm in the UK where the current timezone is GMT, which equates to UTC. Why am I losing an hour somewhere?

**Comment by rehan509** [comment]
Any Update on this problem

**Comment by rehan509** [comment]
@raulbalestra  This error still exist in v5

**Comment by gregg-cbs** [comment]
@rehan509 lol, look like you are stuck with this. I left strapi for vanilla nodejs. Sorry man, hope a solution comes around at some point. For now just store two fields per date, the strapi date and a field must have the timezone.You will have to convert the time to timezone on the server or UI.

I do that in my new nodejs project anyways.
